source/h2_decoder.c

```diff
@@ -633,14 +729,19 @@ static int s_state_fn_frame_window_update(struct aws_h2_decoder *decoder, struct
     AWS_ASSERT(succ);
     (void)succ;
 
+    if (window_increment & s_u32_top_bit_mask) {
+        /* Top bit of stream ids is reserved */
+        DECODER_LOG(ERROR, decoder, "Received WINDOW_UPDATE frame with window size increment with top bit set");
+        return aws_raise_error(AWS_ERROR_HTTP_PROTOCOL_ERROR);
```

Comment:
The WINDOW_UPDATE docs don't say what the rules are with the reserved bit. I'm going to assume rules are the same as the reserved bit in stream-id